vllm/model_executor/layers/rotary_embedding/base.py

```diff
@@ -103,12 +112,131 @@ def forward_cuda(
             self.cos_sin_cache = self.cos_sin_cache.to(query.device,
                                                        dtype=query.dtype)
 
-        # ops.rotary_embedding() is an in-place operation
-        # that updates the query and key tensors.
-        ops.rotary_embedding(positions, query, key, self.head_size,
-                             self.cos_sin_cache, self.is_neox_style)
+        num_tokens = positions.numel()
+        if envs.VLLM_ROCM_USE_AITER and \
+            envs.VLLM_USE_AITER_TRITON_ROPE and num_tokens <= 128:
+
+            import aiter.ops.triton.rope as ops
+            assert key is not None
+
+            cos, sin = self.cos_sin_cache.chunk(2, dim=-1)
+            query_shape = query.shape
+            key_shape = key.shape
+            query = query.view(num_tokens, -1, self.head_size)
+            key = key.view(num_tokens, -1, self.head_size)
+            query_ = query[..., :self.rotary_dim]
+            key_ = key[..., :self.rotary_dim]
+            if offsets is not None:
```

Comment:
Would it be easier to use 
rocm_aiter_rotary_emb_with_key_forward_hip_impl
in rocm_aiter_rope_ops.py?  where we can pass offsets, whether it's none or not without having to check here